src/libs/actions/BankAccounts.js

```diff
@@ -1,11 +1,128 @@
 }
 
+/**
+ * Gets the Plaid Link token used to initialize the Plaid SDK
+ */
+function fetchPlaidLinkToken() {
+    API.Plaid_GetLinkToken()
+        .then((response) => {
+            if (response.jsonCode !== 200) {
+                return;
+            }
+
+            Onyx.merge(ONYXKEYS.PLAID_LINK_TOKEN, response.linkToken);
+        });
+}
+
+/**
+ * List of bank accounts. This data should not be stored in Onyx since it contains unmasked PANs.
+ *
+ * @private
+ */
+let plaidBankAccounts = [];
+let bankName = '';
+let plaidAccessToken = '';
+
+/**
+ * @param {String} publicToken
+ * @param {String} bank
+ */
+function getPlaidBankAccounts(publicToken, bank) {
+    bankName = bank;
+
+    Onyx.merge(ONYXKEYS.PLAID_BANK_ACCOUNTS, {loading: true});
+    API.BankAccount_Get({
+        publicToken,
+        allowDebit: false,
+        bank,
+    })
+        .then((response) => {
+            plaidAccessToken = response.plaidAccessToken;
+            plaidBankAccounts = response.accounts;
+            Onyx.merge(ONYXKEYS.PLAID_BANK_ACCOUNTS, {
+                error: {
+                    title: response.title,
+                    message: response.message,
+                },
+                loading: false,
+                accounts: _.map(plaidBankAccounts, account => ({
+                    ...account,
+                    accountNumber: Str.maskPAN(account.accountNumber),
+                })),
+            });
+        });
+}
+
+/**
+ * We clear these out of storage once we are done with them so the user must re-enter Plaid credentials upon returning.
+ */
+function clearPlaidBankAccounts() {
+    plaidBankAccounts = [];
+    bankName = '';
+    Onyx.set(ONYXKEYS.PLAID_BANK_ACCOUNTS, {});
+}
+
+/**
+ * Adds a bank account via Plaid
+ *
+ * @param {Object} account
+ * @param {String} password
+ * @param {String} plaidLinkToken
+ */
+function addPlaidBankAccount(account, password, plaidLinkToken) {
+    const unmaskedAccount = _.find(plaidBankAccounts, bankAccount => (
+        bankAccount.plaidAccountID === account.plaidAccountID
+    ));
+    API.BankAccount_Create({
+        accountNumber: unmaskedAccount.accountNumber,
+        addressName: unmaskedAccount.addressName,
+        allowDebit: false,
+        confirm: false,
+        isSavings: unmaskedAccount.isSavings,
+        password,
+        routingNumber: unmaskedAccount.routingNumber,
+        setupType: 'plaid',
+        additionalData: JSON.stringify({
+            useOnFido: false,
+            policyID: '',
+            plaidLinkToken,
+            isInSetup: true,
+            bankAccountInReview: null,
+            currentStep: 'AccountOwnerInformationStep',
+            bankName,
+            plaidAccountID: unmaskedAccount.plaidAccountID,
+            ownershipType: '',
+            acceptTerms: true,
+            country: 'US',
+            currency: 'USD',
+            fieldsType: 'local',
+            plaidAccessToken,
+            isForStripeConnect: false,
```

Comment:
@ctkochan22 wondering if you can help us figure out whether we need all of this stuff. I loosely based these parameters on what we send when adding a PBA on e.com.

Reply:
Don't need `isForStripeConnect`. The rest are necessary I think. But when it comes to additionalData, more info is always good I think. What you have now is good.